src/exports.cc

```diff
@@ -65,16 +65,22 @@ Pairs toPairs(std::string_view buffer) {
   if (buffer.size() < sizeof(uint32_t)) {
     return {};
   }
-  auto size = wasmtoh(*reinterpret_cast<const uint32_t *>(b));
+  bool reverse = "null" != contextOrEffectiveContext()->wasmVm()->getEngineName();
```

Comment:
I believe this is the host pass of Wasm plugins... so I wouldn't do this here multiple times at least.